Кулешова Ирина

trtz opened the pull request.

trtz committed:
первая попытка

trtz committed:
первая попытка

trtz committed:
первая попытка

trtz committed:
Update lego.js

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

trtz committed:
Update lego.js

trtz committed:
Update index.js

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

trtz committed:
Update lego.js

honest-hrundel commented:
:tomato: Пройдено тестов 13 из 15

trtz committed:
Update lego.js

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

trtz committed:
add, or, priorities

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

trtz committed:
линтинг...

honest-hrundel commented:
:tomato: Пройдено тестов 13 из 15

trtz committed:
copy in formatIn

honest-hrundel commented:
:tomato: Пройдено тестов 13 из 15

trtz committed:
Update lego.js

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

trtz committed:
format

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

trtz committed:
Update lego.js

honest-hrundel commented:
:green_apple: Пройдено тестов 18 из 18

honest-hrundel commented:
@bazhenova обрати внимание решено доп. задание

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,179 @@
  */
 exports.isStar = true;
 
+var PRIORITIES = {
```
Хм, какой интересный объект. А можно поподробнее, что за приоритеты?

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,179 @@
  */
 exports.select = function () {
-    return;
+    var properties = getValues(arguments);
```
`properties` -> `fields`, и если я правильно поняла, то тут можно сделать так:

``` js
var fields = [].slice.call(arguments);
```

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,179 @@
+    var properties = getValues(arguments);
+
+    return function select(collection) {
+        return collection.map(function (el) {
```
Давай будем чуть-чуть по-другому сокращать, чтобы было более понятно. Например, тут `el` -> `elem`

bazhenova replied:
Пройдись также по всем аналогичным местам в файле

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,179 @@
  */
 exports.sortBy = function (property, order) {
-    console.info(property, order);
+    order = (order === 'asc') ? 1 : -1;
```
Давай не будем менять переданные параметры, для этого лучше создавать отдельные переменные

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,179 @@
-
-    return;
+    return function limit(collection) {
+        return collection.slice(0, count);
```
Если сюда передать отрицательное число, результат будет неожиданным :) Необходима проверка.

``` js
[1,2,3,4].slice(0, -1)
// [1, 2, 3]
```

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,179 @@
      */
     exports.or = function () {
-        return;
+        var functions = getValues(arguments);
```
Может быть тут тоже подойдет:

``` js
var functions = [].slice.call(arguments);
```

bazhenova replied:
А также в методе `and`

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,179 @@
  */
 exports.query = function (collection) {
-    return collection;
+    var functions = getValues(arguments);
```
И если я правильно поняла, то тут тоже можно воспользоваться методом slice:

``` js
var functions = [].slice.call(arguments);
```

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,179 @@
+    var functions = getValues(arguments);
+    functions.splice(0, 1);
+    var copied = copyCollection(collection);
+    functions.sort(function (fFirst, fSecond) {
```
Может все же читабельнее будет `fFirst` -> `func1`, `fSecond` -> `func2`

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,179 @@
+    functions.splice(0, 1);
+    var copied = copyCollection(collection);
+    functions.sort(function (fFirst, fSecond) {
+        if (PRIORITIES[fFirst.name] > PRIORITIES[fSecond.name]) {
```
Когда используешь какую-нибудь модную штуку типа имени функции, надо всегда помнить, что не все браузеры могут её поддерживать. В данном случае имя функции из спецификации es6, и поддержка у нее так себе. Это тебе на будущее, тут можно не исправлять

bazhenova commented:
🍅

trtz committed:
исправлено:

более понятные названия, ошибка при отрицательной границе, [].slice.call(arguments)

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

trtz committed:
линтинг

honest-hrundel commented:
:green_apple: Пройдено тестов 18 из 18

honest-hrundel commented:
@bazhenova обрати внимание решено доп. задание

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,175 @@
 exports.query = function (collection) {
-    return collection;
+    var functions = [].slice.call(arguments);
+    functions.splice(0, 1);
```
можно сразу выбрать функции из аргументов так:

``` js
var functions = [].slice.call(arguments, 1);
```

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,175 @@
+            return 0;
+        }
+
+        return -1;
```
можно второй if объединить с этим `return` (подсказка: использовать тернарный оператор)

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,175 @@
+        return copyCollection(collection)
+            .sort(function (elem1, elem2) {
+                return (elem1[property] > elem2[property] ? 1 : -1) *
+                (order === 'asc' ? 1 : -1);
```
Лучше так не делать, код становится менее читабельным. У тебя же был хороший вариант до этого, просто надо было не перезаписывать переданный параметр, а создать новую переменную и в нее записать результат тернарного оператора

bazhenova commented on lego.js:
```diff
@@ -6,84 +6,175 @@
 exports.limit = function (count) {
-    console.info(count);
+    if (count < 0) {
+        throw new RangeError('Count must be >= 0!');
```
Можно было просто перезаписать `count` в данном случае, например, сделать его 0. А так получается, что ты кидаешь ошибку, то есть будет прекращено выполнение программы, хотя такие ошибки можно безболезненно обработать

bazhenova commented:
🍅

trtz committed:
done!

honest-hrundel commented:
:green_apple: Пройдено тестов 18 из 18

honest-hrundel commented:
@bazhenova обрати внимание решено доп. задание

trtz commented:
🍏

bazhenova commented:
🚀

mokhov commented on lego.js:
```diff
@@ -6,84 +6,173 @@
+    var functions = [].slice.call(arguments, 1);
+    var copied = copyCollection(collection);
+    functions.sort(function (func1, func2) {
+        if (PERFORMING_PRIORITIES[func1.name] > PERFORMING_PRIORITIES[func2.name]) {
```
Здесь можно воспользоваться `Math.sign`
